hub/core/transform/transform_shard.py

```diff
@@ -0,0 +1,36 @@
+from hub.core.transform.transform_tensor import TransformDatasetTensor
+from hub.util.exceptions import TensorDoesNotExistError
+
+# TODO: add tests
+
+
+class TransformDatasetShard:
+    def __init__(self, all_tensors=None, slice_list=None):
+        self.tensors = all_tensors or {}
+        self.slice_list = slice_list or []
+
+    def __len__(self):
+        return min(len(self[tensor]) for tensor in self.tensors)
```

Comment:
Shouldn't `slice_list` be taken into account here?

Reply:
the __getitem__ will already take it into account